src/RepoAuditor/Plugins/GitHub/Impl/Common.py

```diff
@@ -6,13 +6,31 @@
 # -------------------------------------------------------------------------------
 """Contains common functionality that is used across different components."""
 
+from typing import Optional
+
 from RepoAuditor.Requirement import EvaluateResult, Requirement
 
 
 # ----------------------------------------------------------------------
-def CreateIncompleteDataResult() -> Requirement.EvaluateImplResult:
-    """Create an incomplete data result."""
+def CreateIncompleteDataResult(pat_value: Optional[str]) -> Requirement.EvaluateImplResult:

```

Comment:
It looks like this function doesn't care about the pat value itself, but rather if the pat value exists or not. As such, it should probably be `has_pat_value: bool` rather than `pat_value: Optional[str]`. Invoking the function is a bit more cumbersome, but I'd argue is more correct.

Reply:
Wouldn't that cause `ruff` to [complain about positional boolean arguments](https://docs.astral.sh/ruff/rules/boolean-type-hint-positional-argument/)? I guess we can just ignore it but it indicates poor style(?).